src/Snippets/Docs/Chaos.Outcome.cs

```diff
@@ -105,15 +106,138 @@ public static void OutcomeGeneratorDelegates()
+                        < 250 => Outcome.FromResult(CreateResultFromContext(args.Context)),
+                        < 350 => Outcome.FromException<HttpResponseMessage>(new HttpRequestException("Chaos request exception.")),
+                        _ => Outcome.FromResult(new HttpResponseMessage(HttpStatusCode.OK))
+                    };
+
+                    return ValueTask.FromResult(outcome);
+                },
+                OnOutcomeInjected = static args =>
+                {
+                    // You might have to put some logic here to determine what kind of output was injected. 😕
+                    if (args.Outcome.Exception != null)
+                    {
+                        Console.WriteLine($"OnBehaviorInjected, Exception: {args.Outcome.Exception.Message}, Operation: {args.Context.OperationKey}.");
+                    }
+                    else
+                    {
+                        Console.WriteLine($"OnBehaviorInjected, Outcome: {args.Outcome.Result}, Operation: {args.Context.OperationKey}.");
+                    }
+
+                    return default;
+                }
+            })
+            .Build();
+
+        #endregion
+    }
+
+    public static void Pattern_GeneratorDelegateInjectFaultAndOutcome()
+    {
+        #region chaos-outcome-pattern-generator-inject-fault
+        var randomThreshold = Random.Shared.Next(350);
+        var pipeline = new ResiliencePipelineBuilder<HttpResponseMessage>()
+            .AddChaosFault(new ChaosFaultStrategyOptions
+            {
+                InjectionRate = 0.1, // Different injection rate for faults
+                EnabledGenerator = static args => ValueTask.FromResult(ShouldEnableFaults(args.Context)), // Different settings might apply to inject faults
+                FaultGenerator = args =>
+                {
+                    Exception? exception = randomThreshold switch
+                    {
+                        >= 250 and < 350 => new HttpRequestException("Chaos request exception."),
                         _ => null
                     };
 
+                    return new ValueTask<Exception?>(exception);
+                },
+                OnFaultInjected = static args =>
+                {
+                    Console.WriteLine($"OnFaultInjected, Exception: {args.Fault.Message}, Operation: {args.Context.OperationKey}.");
+                    return default;
+                }
+            })
+            .AddChaosOutcome(new ChaosOutcomeStrategyOptions<HttpResponseMessage>
+            {
+                InjectionRate = 0.5, // Different injection rate for outcomes
+                EnabledGenerator = static args => ValueTask.FromResult(ShouldEnableOutcome(args.Context)), // Different settings might apply to inject outcomes
+                OutcomeGenerator = args =>
+                {
+                    Outcome<HttpResponseMessage>? outcome = randomThreshold switch
```

Comment:
In order to reduce repetitive code the `switch` expression should either return a `HttpStatusCode` or an `HttpResponseMessage`. Something like this:

```cs
HttpStatusCode statusCode = randomThreshold switch
{
    < 100 => HttpStatusCode.InternalServerError,
    < 150 => HttpStatusCode.TooManyRequests,
    < 250 => CalculateStatusCodeFromContext(args.Context),
    _ => HttpStatusCode.OK
};

var outcome = Outcome.FromResult(new HttpResponseMessage(statusCode));
return ValueTask.FromResult(outcome);
```